packages/vite/src/node/server/hmr.ts

```diff
@@ -62,9 +62,14 @@ export async function handleHMRUpdate(
   const isConfigDependency = config.configFileDependencies.some(
     (name) => file === name,
   )
+  const isDev = config.mode === 'development'
+  const isChangedProductionEnv =
+    isDev &&
+    (fileName === '.env.production' || fileName === '.env.production.local')
```

Comment:
Is there some part of the code where we already build the list of files for a given mode? We could use this to be more precise instead a special case here

Reply:
I didn't find the relevant code, can you give me a hint?

Reply:
Maybe I just need to make the following changes?
```ts
function changeEnvShouldRestart(mode: string, fileName: string) {
  if (fileName === '.env' || fileName === '.env.local') {
    return true
  }
  return fileName === `.env.${mode}` || fileName === `.env.${mode}.local`
}
......
const isEnv =
    config.inlineConfig.envFile !== false && changeEnvShouldRestart(config.mode, fileName)
```

Reply:
I don't where it is but this code definitely better 👌